includes/import_to_drupal.inc

```diff
@@ -237,8 +231,34 @@ function islandora_sync_save_field($field_info, &$entity, $value) {
       module_load_include('inc', 'islandora', 'mime_detect');
       $mime_detect = new MimeDetect();
       $extension = $mime_detect->getExtension($value['mimetype']);
-      $file_name = $field_info->dsid . '-' . $entity->type . '-' . $id . '.' . $extension;
-      $file_uri = "public://islandora_sync/$id/$file_name";
+      // Retrieve the settings for this field.
+      $field_settings = field_info_instance('node', $field_info->field, $entity->type);
+
+      // Determine if the field configuration has already
+      // specified a subdirectory where files should be saved.
+      if ((isset($field_settings['settings']['file_directory'])) &&
+          (trim($field_settings['settings']['file_directory']) != "")) {
+        $file_name_component = NULL;
+        // Use subdirectory path already specified in the field configuration.
+        $subdirectory = $field_settings['settings']['file_directory'];
+        $node_path = "public://$subdirectory";
+        if (strtolower($field_info->dsid) != $extension) {
+          $file_name_component = '-' . str_replace($extension, '', strtolower($field_info->dsid));
+        }
```

Comment:
@slangerx, I'm not clear why this is needed, but if it is why does it only appears when there is a file_directory defined. Should it not appear below in the ```else``` as well?